### DISCUSSION TITLE
Installing Cardinal from "scratch "

### DISCUSSION THREAD

**Discussion by AnthonyB08** [neams-th-coe/cardinal#574] [Q&A] [answered]
Hi all new Cardinal user,

From the documentation it is said that for NEKS, OpenMC, and Moose to all be coupled, Cardinal must be built outside of MOOSE conda environment, correct me if I am wrong. So does this mean that Moose must also be built from scratch to work coherently with NEKS and OpenMC.

For context of why I want NEKS is that I am trying to conduct research in CFD and Neutronics. In particular, I am interested in turbulent fluid flow in a reactor. So I am thinking I will need to include NEKS with OpenMc and MOOSE, is this correct?  

Additionally, I'd like to ask if there are any efforts being made to include a two-phase interactions module in the future for research regarding species removal.

**Comment by aprilnovak** [comment]
Welcome @AnthonyB08! Yes, if you want to run any cases that use NekRS, then the MOOSE conda environment has to be disabled. This isn't too onerous - at a high level, instead of just loading the conda environment and building, like so:

```
conda activate moose
make
```

you instead just need to build PETsc and libMesh first, like so:

```
./contrib/moose/scripts/update_and_rebuild_petsc.sh
./contrib/moose/scripts/update_and_rebuild_libmesh.sh
make
```

It'll take longer to build, but should be equivalent. Full instructions are here: https://cardinal.cels.anl.gov/without_conda.html

**Comment by aprilnovak** [comment]
And yes, if you would like to model turbulence at high-fidelity, you will need to use a CFD solver. NekRS is highly performant and has a wide variety of turbulence options (RANS, LES, and DNS), so is a good choice for many reactor applications. 

I do not believe there are any plans to have two-phase flow capabilities in the near future (e.g. liquid-vapor systems). But you can use NekRS for chemical species transfer (where each mass species is represented as a passive scalar). Can you tell me a bit more about your application? I may be able to provide more details or point you to specific documentation/examples.

**Reply by AnthonyB08** [reply]
My undergade/grad research investigated the removal of tritium and xenon with helium through the mechanics of sparging. I don't think this will fully be capable in Cardinal as of now, but I am willing to work towards a solution. For now, I am interested in a simple neutronics+heat+fluid flow simulation so I have some multiphysics experience under my belt. 

Thank you for you quick response

**Reply by AnthonyB08** [reply]
I should also add that the majority of my research involved CFD analysis using Fluent. So using this open-sourced CFD will be a learning curve - but an enjoyable one.

**Reply by aprilnovak** [reply]
Sounds like an interesting application. Using NekRS will feel different from a GUI-based CFD solver, but I'm happy to help with any questions you may have. Some resources:

- NekRS documentation: https://nekrsdoc.readthedocs.io/en/latest/index.html
- NekRS discussions page: https://github.com/Nek5000/nekRS/discussions

We also have slack spaces for Nek and Cardinal.

**Comment by aprilnovak** [comment, marked as answer]
Closing thread because a fix was found at: https://github.com/neams-th-coe/cardinal/discussions/589

**Comment by emerzari** [comment]
Anthony, what kind of two-phase modeling do you have in mind? We have plans but it does really depend a lot on what you are thinking.
If you want something Lagrangian it may not be a huge lift.
You can reach out to me at ***@***.*** if you want.

**Reply by aprilnovak** [reply]
Did github censor the email? I just see ***@***.***

**Reply by AnthonyB08** [reply]
@emerzari  Git hub did censor the email, haha! I am interested in modeling an eulerian-eulerian environment with species transport, i.e tritium from the liquid phase to the gas phase.  A lagrangian model will be too computationally expensive for my case.

 If GitHub does not censor this link , haha,  we keep In touch via LinkedIn. Once we connect, I can send the email since github is censoring our contact info. 

https://www.linkedin.com/in/anthony-bowers-372236127/
